src/reconcile/rec_row.c

```diff
@@ -1042,6 +1044,8 @@ __wt_rec_row_leaf(
     ret = __wt_rec_split_finish(session, r);
 
 err:
+    if (hs_cursor != NULL)
+        WT_RET(hs_cursor->close(hs_cursor));
```

Comment:
I think we need WT_TRET here, since if it doesn't close properly, then it will just return without freeing the buffer

Reply:
Agree

Reply:
Fixed.